sqlmesh/core/snapshot/evaluator.py

```diff
@@ -256,23 +264,12 @@ def demote(
                 self.ddl_concurrent_tasks,
             )
 
-    def create(
+    def get_snapshots_to_create(
```

Comment:
I don't like this interface. It reveals too much implementation details of the snapshot evaluator and makes the plan evaluator (or any other calling code) more complex . I suggest instead passing another callback into the `create` method which will be called once the number of snapshots needing creation is known.